Day04/Module11UnderstandingSequelize/index.js

```diff
@@ -0,0 +1,188 @@
+
+// to convert body data into json
+app.use(bodyparse.json());
+
+//app.get('/employee/form',employeeRoutes);
+
+app.get('/',(req,res,next)=>{
+     res.send(`<h1>please use correct URL</h1>`);
+});
+
+app.get('/add',(req,res,next)=>{
+     res.send(`<h2>Submit Form</h2>
+
+     <form action="/employee/add" method="post">
+     <label for="name">Name:</label><br>
+     <input type="text" id="name" name="name"><br>
+     
+     <label for="company">Company:</label><br>
+     <input type="text" id="company" name="company"><br>
+     
+     <label for="location">Location:</label><br>
+     <input type="text" id="location" name="location"><br><br>
+     
+     <input type="submit" value="Submit">
+   </form>`)
+});
+
+
+
+
+app.get('/update',(req,res,next)=>{
+    res.send(`<h2>Submit Form</h2>
+
+    <form action="/employee/update" method="post">
+    <label for="name">Name:</label><br>
+    <input type="text" id="name" name="name"><br>
+    
+    <label for="company">Company:</label><br>
+    <input type="text" id="company" name="company"><br>
+    
+    <label for="location">Location:</label><br>
+    <input type="text" id="location" name="location"><br><br>
+    
+    <input type="submit" value="Submit">
+  </form>`)
+})
+
+
+
+
+
+
+
+// Inserting data in table using sequelize 
+// alway use promises 
+app.get('/employee/show',(req,res,next)=>{
+     Employee.findAll().then((resolve)=>{
+        console.log('get all the data')
+        for(const employee of resolve){
+             console.log(employee);
+        }
+        res.send('<h2>We have get all the details<h2/>')
+     }).catch((reject)=>{
+           console.log("have some",reject);
+     })
+})
+
+app.get('/employee/show/:id',(req,res,next)=>{
+     const employeeid = req.params.id;
+
+     Employee.findAll({where :{id:employeeid}}).then((resolve)=>{
+           console.log(`single data finded by ${employeeid}`,resolve);
+           const {id,name,company,location} = resolve[0];
+           console.log("resolve",resolve[0].id,resolve[0].name,resolve[0].company,resolve[0].location);
+           res.send(`<h1>Employee Details</h1><h3>employeeid : ${resolve[0].id}</h3>
+           <h3>employeename : ${resolve[0].name}</h3>
+           <h3>employeecompany : ${resolve[0].company}</h3>
+           <h3>employeelocation : ${resolve[0].location}</h3>`)
+     }).catch((reject)=>{
+          console.log(reject);
+     })
+})
+
+
+
+app.put('/employee/update/:id',(req,res,next)=>{
+      const employeeid = req.params.id;
+      const employee = req.body;
+       console.log("employee : ",employee);
+       console.log("req.body: ",req.body);
+      Employee.findAll({where :{id:employeeid}}).then((resolve)=>{
+        console.log(`single data finded by ${employeeid}`,resolve);
+        const {id,name,company,location} = resolve[0];
+        console.log("resolve",resolve[0].id,resolve[0].name,resolve[0].company,resolve[0].location);
+
+        resolve[0].name = req.body.name;
```

Comment:
Bad practice doing this, check the methods that sequelize provides to interact with models for CRUD
at this case we can use update